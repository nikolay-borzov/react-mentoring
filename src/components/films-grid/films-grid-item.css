@import '../../styles/variables.css';

.film-grid-item {
  display: flex;
  flex-direction: column;
  color: var(--color-default);
  text-decoration: none;

  @apply --hover-effect;
}

.film-grid-item__cover {
  max-width: 100%;
  display: block;
}

.film-grid-item__info {
  margin-top: 0.5em;
}

.film-grid-item__row {
  padding: 0.4em 0.5em;
}

.film-grid-item__title {
  display: flex;
  justify-content: space-between;
  align-items: start;
}

.film-grid-item__date {
  border: solid 1px var(--color-alt);
  padding: 0.1em 0.5em;
  border-radius: 0.2em;
  margin-left: 1em;
}
