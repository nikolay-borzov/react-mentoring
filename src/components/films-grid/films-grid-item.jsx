// @flow

import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import { ContentImage } from '../'

type FilmsGridItemProps = {
  film: Film
}

const FilmGridItem = styled(Link)`
  display: flex;
  flex-direction: column;
  color: var(--color-default);
  text-decoration: none;
`

const FilmGridItemInfo = styled.div`
  margin-top: 0.5rem;
`

const FilmGridItemRow = styled.div`
  padding: 0.4rem 0.5rem;
`

const FilmGridItemTitle = styled(FilmGridItemRow)`
  display: flex;
  justify-content: space-between;
  align-items: start;
`

const FilmGridItemDate = styled.span`
  border: solid 1px var(--color-alt);
  padding: 0.1rem 0.5rem;
  border-radius: 0.2rem;
  margin-left: 1rem;
`

export function FilmsGridItem(props: FilmsGridItemProps) {
  const {
    film: {
      id,
      poster_path: posterPath,
      title,
      overview,
      release_date: releaseDate
    }
  } = props

  return (
    <FilmGridItem
      to={`/film/${id}`}
      className="hover-effect"
      tabIndex={0}
      data-cy="film-grid-item">
      <ContentImage src={posterPath} alt={title} title={overview} />

      <FilmGridItemInfo>
        <FilmGridItemTitle>
          <span className="uppercase font-bold">{title}</span>
          <FilmGridItemDate>{releaseDate.substring(0, 4)}</FilmGridItemDate>
        </FilmGridItemTitle>

        <FilmGridItemRow className="color-alt">
          {props.film.genres.join(', ')}
        </FilmGridItemRow>
      </FilmGridItemInfo>
    </FilmGridItem>
  )
}
