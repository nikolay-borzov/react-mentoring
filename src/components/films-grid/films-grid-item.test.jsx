import React from 'react'

import {
  itRendersCorrectly,
  itContainsComponent
} from '../../../jest/test-helpers'

import { FilmsGridItem } from './films-grid-item'

describe('FilmsGridItem component', () => {
  let props = {
    film: {
      id: 1,
      title: 'Film title',
      poster_path: '//content/posters/1.png',
      overview: 'Film overview',
      release_date: '2018-05-04',
      genres: ['some genre', 'another genre']
    }
  }

  itRendersCorrectly(() => <FilmsGridItem {...props} />)

  itContainsComponent(() => <FilmsGridItem {...props} />, 'ContentImage', {
    expectedProps: {
      src: props.film.poster_path,
      alt: props.film.title,
      title: props.film.overview
    }
  })
})
