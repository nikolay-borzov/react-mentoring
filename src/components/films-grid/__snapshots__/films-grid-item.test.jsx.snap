// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsGridItem component renders correctly 1`] = `
<a
  className="film-grid-item hover-effect"
  data-cy="film-grid-item"
  href="?id=1"
  tabIndex={0}
>
  <ContentImage
    alt="Film title"
    src="http://via.placeholder.com/500x750"
    title="Film overview"
  />
  <div
    className="film-grid-item__info"
  >
    <div
      className="film-grid-item__row film-grid-item__title"
    >
      <span
        className="uppercase font-bold"
      >
        Film title
      </span>
      <span
        className="film-grid-item__date font-size-small color-alt"
      >
        2018
      </span>
    </div>
    <div
      className="film-grid-item__row color-alt"
    >
      some genre, another genre
    </div>
  </div>
</a>
`;
